project/FrgaalJavaCompiler.scala

```diff
@@ -63,6 +64,58 @@ object FrgaalJavaCompiler {
     val sources = sources0 map {
       case x: PathBasedFile => x.toPath.toAbsolutePath.toString
     }
+    val out = output.getSingleOutputAsPath().get()
+    val shared = sources0.fold(out)((a, b) => {
```

Comment:
So this tries to find the top common parent between the sources?

Reply:
Yes. Between all the sources and the output directory.